src/native/crt.c

```diff
@@ -202,7 +203,20 @@ struct aws_string *aws_jni_new_string_from_jstring(JNIEnv *env, jstring str) {
     return aws_string_new_from_c_str(allocator, (*env)->GetStringUTFChars(env, str, NULL));
 }
 
+void s_detach_jvm_from_thread(void *user_data) {
+    JavaVM *jvm = user_data;
+    (*jvm)->DetachCurrentThread(jvm);
+}
+
+void s_attach_jvm_cleanup_to_thread(void *user_data) {
+    AWS_FATAL_ASSERT(AWS_OP_SUCCESS == aws_thread_current_at_exit(s_detach_jvm_from_thread, user_data));
+}
+
+static aws_thread_once s_jvm_once = AWS_THREAD_ONCE_STATIC_INIT;
+
 JNIEnv *aws_jni_get_thread_env(JavaVM *jvm) {
+    aws_thread_call_once(&s_jvm_once, s_attach_jvm_cleanup_to_thread, jvm);
```

Comment:
I'm pretty sure I wrote this to emulate std::call_once, which means it gets called once per process, not once per thread. Is that intended?

Reply:
Nope, I was so annoyed at LTO I forgot how to computer, will fix